src/client.js

```diff
@@ -5,23 +5,51 @@ const url = require("url");
 
 // Client can read/write messages from a TCP server
 class Client {
-  // init initializes the Client
-  init(addr) {
-    let u = url.parse("tcp://" + addr, false, false);
-    this.socket = new net.Socket();
-    this.socket.connect(u.port, u.hostname, function() {});
-    this.socket.on("close", function() {
-      process.exit();
-    });
-    return this;
-  }
-
-  // write writes an event to the server
-  write(targetID, eventName, payload) {
-    let data = { name: eventName, targetID: targetID };
-    if (typeof payload !== "undefined") Object.assign(data, payload);
-    this.socket.write(JSON.stringify(data) + "\n");
-  }
+    // init initializes the Client
+    init(addr) {
+
+        this.connect(addr)
+
+        this.socket.on('error', function(err){
+            // Writing to a file in case of error related to socket
```

Comment:
In this case I'd rather throw an exception (that will be printed on the user's screen) and kill the app instead of writing to a file few people are aware of.

I don't know how Electron does it and I'm not really familiar with NodeJS, but when there's a fatal error in Electron, it prints the error in a nice window and exit the process. Could you do the same thing instead?

If this is not possible, you should only have to print the error in `stderr` which will then be picked up by the `go` side that will print it in its logs.